common/src/main/scala/org/apache/celeborn/common/util/PbSerDeUtils.scala

```diff
@@ -446,9 +448,24 @@ object PbSerDeUtils {
       builder.setCurrentAppDiskUsageMetricsSnapshot(
         toPbAppDiskUsageSnapshot(currentAppDiskUsageMetricsSnapshot))
     }
+    if (localCollectionUtils.isNotEmpty(applicationMetas)) {
+      builder.putAllApplicationMetas(applicationMetas.asScala.map {
+        case (appId, applicationMeta) => (appId, toPbApplicationMeta(applicationMeta))
+      }.asJava)
+    }
```

Comment:
Can `applicationMetas` be concurrently modified ? If yes:
```suggestion
    val pbApplicationMetas = applicationMetas.asScala.map {
        case (appId, applicationMeta) => (appId, toPbApplicationMeta(applicationMeta))
      }.asJava
    if (localCollectionUtils.isNotEmpty(pbApplicationMetas)) {
      builder.putAllApplicationMetas(pbApplicationMetas)
    }
```

Reply:
Fixed with other changes